client/state/sites/selectors.js

```diff
@@ -58,12 +58,12 @@ export const getRawSite = ( state, siteId ) => {
 export const getSiteBySlug = createSelector(
 	( state, siteSlug ) =>
 		find(
-			state.sites.items,
+			getSitesItems( state ),
 			( item, siteId ) =>
 				// find always passes the siteId as a string. We need it as a integer
 				getSiteSlug( state, parseInt( siteId, 10 ) ) === siteSlug
 		) || null,
```

Comment:
nit: `find` will already return `null` if the object to search in is `null` or if nothing is found. The `|| null` bit is redundant.

Reply:
lodash docs say that `find()` returns `undefined` if the item isn't found

Reply:
That's totally right, both for `get` and `find`. It was some hallucination of mine being convinced they return `null` 😄

Reply:
playtime:

```js
function js(undefined) {
  return undefined % 10
}

js( 13 ) === 3
```

because why the heck not?